### ISSUE TITLE
SDL2 wayland support appears broken

### ISSUE THREAD

**Issue by mdorman** [NixOS/nixpkgs#31895]
## Issue description

SDL2 fails to build, complaining of a missing wayland header.  This in turn, breaks ffmpeg3, which breaks a whole host of other stuff.  @gnidorah, I'm *guessing* that your b0d193f596d is the culprit, just because it added the wayland flag to SDL2; that may not be correct, but I'm hoping you're at least in a good position to fix the issue.

### Steps to reproduce

https://hydra.nixos.org/build/64495513/nixlog/1

**Comment by gnidorah** [comment]
Should be temporary fixed by https://github.com/NixOS/nixpkgs/commit/3772338be5b9ffbbc6b1b93bbd6ea052533a6180
Will see if it could be fixed in a bettery way. The error means that header is not yet generated from xml file at the stage it got requested.

**Comment by vcunat** [comment]
The under-specified makefile dependency could be reported upstream.

**Comment by gnidorah** [comment]
Indeed. Reported upstream https://bugzilla.libsdl.org/show_bug.cgi?id=3977
